nifi-toolkit/nifi-toolkit-encrypt-config/src/main/groovy/org/apache/nifi/properties/ConfigEncryptionTool.groovy

```diff
@@ -1269,10 +1273,10 @@ class ConfigEncryptionTool {
     }
 
     private
-    static List<String> serializeNiFiPropertiesAndPreserveFormat(NiFiProperties niFiProperties, File originalPropertiesFile) {
+    static List<String> serializeNiFiPropertiesAndPreserveFormat(NiFiProperties niFiProperties, File originalPropertiesFile, String keyHex) {
```

Comment:
This would change to a provider instance due to the change in `ProtectedNiFiProperties` constructors.

Reply:
When I first refactored the String keys into `SensitivePropertyProvider` instances, it made sense to adjust the method signatures.  However, after reviewing the net change, most of the calls seemed like they had been over complicated with little to no benefit to the caller.

So I re-re-factored (as I mentioned above) to allow clients to pass in string keys.  In all of those cases, the client was still maintaining the string key anyway, so it's not as if we've added any new un-protected key usage.